components/sim800l.rst

```diff
@@ -0,0 +1,112 @@
+Sim800L Component
+=================
+
+.. seo::
+    :description: Instructions for setting up the SIM800L gsm module to send and receive SMS in ESPHome.
+    :image: sim800l.png
+    :keywords: SMS SIM800L GSM
+
+The ``SIM800L`` Component provides the ability to send and receive SMS text messages. The device must be
+connected via a :doc:`UART bus </components/uart>` supporting both receiving and transmitting line.
+
+The uart bus must be configured at the same speed of the module which is by default 9600bps. Either hardware
+or software uarts are supported. If you are using the hardware uart in the ESP8266, make sure you disable
```

Comment:
I just want to add all most relevant information that might cause trouble. By saying this I am also saying that the TX / RX lines can be used for Sim800L (some user may believe it can only be used for logging)
May be adding another `.. note::` / `.. warning::` about using the `sim800l` and `logger` component at the same time.
I did not test it, but I guess it will fail if you don't explicitly disable the logger via serial. 
About the connections: looks like a good idea, there are some other pins which don't need to be connected and putting it clear will answer questions before they are made, like reset, it also has pins for speaker, mic, etc.
I'll push more later...